### PR TITLE
Create Other Instrumentation Mapper

### PR DESCRIPTION
**Pull request by heyagosper**

### CODE REVIEW

File: src/main/java/au/gov/ga/geodesy/domain/model/sitelog/EffectiveDates.java [hunk 1]
```diff
@@ -67,4 +67,24 @@ public boolean equals(Object x) {
         }
         return from.equals(other.getFrom()) && to.equals(other.getTo());
     }
+
+    /**
+     * Implement compareTo method so that collections of EffectiveDate objects can be sorted properly
+     * @param other the EffectiveDate to compare with this one
+     * @return int value of comparison
+     */
+    public int compareTo(EffectiveDates other) {
```
Comment (oehm-smith) [inline src/main/java/au/gov/ga/geodesy/domain/model/sitelog/EffectiveDates.java]: You need to `implement Comparable`

Comment (oehm-smith) [inline src/main/java/au/gov/ga/geodesy/domain/model/sitelog/EffectiveDates.java]: And so how about a test for that also?

File: src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/OtherInstrumentationMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        OtherInstrumentationLogItem logItem = mapper.to(otherInstrumentationTypeA);
+        assertEquals(logItem.getInstrumentation(), otherInstrumentationTypeA.getInstrumentation());
+        String xmlEffectiveDateFrom = ((TimePeriodType) otherInstrumentationTypeA.getValidTime().getAbstractTimePrimitive().getValue())
+                .getBeginPosition().getValue().get(0);
+        assertEquals(GMLDateUtils.dateToString(logItem.getEffectiveDates().getFrom(), GMLDateUtils.GEODESYML_DATE_FORMAT_TIME_MILLISEC),
+                GMLDateUtils.stringToDateToStringMultiParsers(xmlEffectiveDateFrom));
+
+        OtherInstrumentationType otherInstrumentationTypeB = mapper.from(logItem);
+        assertEquals(logItem.getInstrumentation(), otherInstrumentationTypeB.getInstrumentation());
+        xmlEffectiveDateFrom = ((TimePeriodType) otherInstrumentationTypeB.getValidTime().getAbstractTimePrimitive().getValue())
```
Comment (oehm-smith) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/OtherInstrumentationMapperTest.java]: Do EffectiveDate.to as well.

Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/OtherInstrumentationMapperTest.java]: none of our tests are doing that btw

File: src/main/java/au/gov/ga/geodesy/domain/model/sitelog/EffectiveDates.java (hunk 1 above)
Comment (heyagosper) [inline src/main/java/au/gov/ga/geodesy/domain/model/sitelog/EffectiveDates.java]: I think that this is exercised sufficiently by SiteLogMapperTest which sorts the XML types and sorts the Entities - what do you think?

File: src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/OtherInstrumentationMapperTest.java (hunk 2 above)
Comment (oehm-smith) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/OtherInstrumentationMapperTest.java]: Yeah I know!  The thing I'm working on for the Dozer convert is to make sure EDs are coming out correctly.  Probably checking it is a TimePeriodType (or having a cast that will fail if it isn't) is probably sufficient.

File: src/test/java/au/gov/ga/geodesy/domain/model/sitelog/EffectiveDatesComparableTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                new EffectiveDates(
+                        now.minusSeconds(100),
+                        now.minusSeconds(90)),
+                new EffectiveDates(
+                        now.minusSeconds(90),
+                        now.minusSeconds(80)),
+                new EffectiveDates(
+                        now.minusSeconds(80),
+                        null)
+        };
```
Comment (oehm-smith) [inline src/test/java/au/gov/ga/geodesy/domain/model/sitelog/EffectiveDatesComparableTest.java]: Add a `null, null` one also.  It should be handled properly.

**Pull request merged**